src/src_main/crypto_tc.c

```diff
@@ -462,12 +462,14 @@ int32_t Crypto_TC_ApplySecurity(const uint8_t* p_in_frame, const uint16_t in_fra
 
             if(ecs_is_aead_algorithm == CRYPTO_TRUE)
             {
+                crypto_key_t* ek_ring = cryptography_if->get_ek_ring();
```

Comment:
We need to remove get_ek_ring() from the cryptography interface. The cryptography interface itself must be responsible for all key interactions since they work very differently. libgcrypt keymanagement from a key store or internal key ring behaves very differently from KMC's crypto service key interface. Remove this call (and possibly the function from the interface entirely).

Reply:
<not duplicating my comments for all new get_ek_ring() functions in this PR, this applies anywhere get_ek_ring() is used here>

Reply:
I agree that it's a good idea to store the key metadata in the keyring, but we shouldn't rely on the keyring for anything at the core CryptoLib level since only one interface uses the keyring (and even that will become optional when https://github.jpl.nasa.gov/ASEC/AMMOS-CryptoLib/issues/42 is implemented). 

The expected key length should be passed into the functions based on the chosen algorithm. To relieve your concern about this being brittle, you can add a length check against the length passed in vs the actual key length inside the libgcrypt interface implementation. 

Not that the get_acs_algo_keylen doesn't need to be implemented per each interface, it can be a core function since the enums are shared between interfaces and owned by CryptoLib.